### ISSUE TITLE
Is the final concatenation of the question for t+1 time step or t time step in the code?

### ISSUE THREAD

**Issue by fishfishfishfishfishfish** [yxonic/DTransformer#7]
In the model, it is mentioned that the final fusion should be qt+1, but in the code, it seems like only qt is being concatenated and there is no indication of any shift operation.

n=1
query = q_emb[:, n - 1 :, :]
y = self.out(torch.cat([query, h], dim=-1)).squeeze(-1)

The following code also supports this observation. It is supposed to predict the label for t+1 time step (starting from the second label), but the answer for s starts from the first label.

masked_logits = logits[s >= 0]
masked_labels = s[s >= 0].float()
pred_loss = F.binary_cross_entropy_with_logits(
            masked_logits, masked_labels, reduction="mean"
        )
 Are there any details that I may have missed? If you can provide any insights, I would greatly appreciate it.

**Comment by githubzmt** [comment]
May I ask if the above mentioned students have resolved the respective queries? Would like to discuss it with you. I was reading through the code and found the same query as you raised above, "Is the final concatenation of the question for t+1 time step or t time step in the code?". Thank you for taking time out of your busy schedule to answer my question.

**Comment by fishfishfishfishfishfish** [comment]
Your response is very clear, thank you for your explanation!

**Comment by Holmes2002** [comment]
@yxonic @fishfishfishfishfishfish Hi when model predicts score in the next step. Does model need a pseudo label in next step (like 1 or simply 0 default as padding) before inference ?
